packages/type-storage/src/fromMetadata/createFunction.ts

```diff
@@ -32,15 +32,36 @@ export default function createFunction (section: Text | string, method: Text | s
 
   // Get the hashing function
   // FIXME Hash correctly for double map too
-  const hasher = meta.type.isMap
-    ? getHasher(meta.type.asMap.hasher)
-    : getHasher();
+  let hasher: HasherFunction;
+  let key2Hasher: HasherFunction;
+
+  if (meta.type.isDoubleMap) {
+    hasher = getHasher(meta.type.asDoubleMap.hasher);
+    key2Hasher = getHasher(meta.type.asDoubleMap.key2Hasher);
+  } else if (meta.type.isMap) {
+    hasher = getHasher(meta.type.asMap.hasher);
+  } else {
+    hasher = getHasher();
+  }
 
   // Can only have zero or one argument:
   // - storage.balances.freeBalance(address)
   // - storage.timestamp.blockPeriod()
   const storageFn = (arg?: any): Uint8Array => {
     let key = rawKey;
+
+    if (meta.type.isDoubleMap) {
+      assert(!isUndefined(arg) && !isNull(arg) && !isUndefined(arg[0]) && !isNull(arg[0]) && !isUndefined(arg[1]) && !isNull(arg[1]), `${meta.name} expects two arguments`);
+      const type1 = meta.type.asDoubleMap.key1.toString();
+      const type2 = meta.type.asDoubleMap.key2.toString();
+
+      const param1Encoded = u8aConcat(key, createType(type1, arg[0]).toU8a(true));
```

Comment:
`true` here means we don't length-prefix the U8a. Intuitively I would say we need to add the length prefix, but I may be wrong.

Would like to see an e2e test (with event_topics)

Reply:
yep. will double check in the e2e test